### PR TITLE
Included support for GLP IAM v1

### PR DESCRIPTION
**Pull request by dbozzato81**
Included support for GLP IAM v1

### CODE REVIEW

File: examples/provider/provider.tf [hunk 1]
```diff
@@ -10,12 +10,23 @@ terraform {
...
+  }
+}
+
+# Example of provider configuration when using GreenLake Platform (GLP) IAM token
+provider "hpegl" {
+  metal {
+    rest_url   = "https://localhost:3002"
+    project_id = "1d96bfbc-9cf0-4268-aac6-ca1c65aca385"
+    glp_workspace = "1a2ba81600dd11efa47076a3447ec4eb"
+    glp_role = "service_platform_owner"
```
Comment (mchuang3) [inline examples/provider/provider.tf]:
```suggestion
    glp_role = "service-platform-owner"
```

File: pkg/configuration/config.go [hunk 2]
```diff
@@ -201,7 +219,7 @@ func NewConfig(portalURL string, opts ...CreateOpt) (*Config, error) {
 
 	if config.useGLToken || config.trf != nil {
 		if err := validateGLConfig(*config); err != nil {
-			return config, fmt.Errorf("configuration error: %v", err)
+			return config, fmt.Errorf("configuration error: %+v", config)
```
Comment (mchuang3) [inline pkg/configuration/config.go:222]: Did you  mean to remove err? Seems like err msg returned would be good. Can add config output as additional data, but err should be included.

File: pkg/registration/registration.go [hunk 3]
```diff
@@ -81,9 +83,21 @@ func (r Registration) ProviderSchemaEntry() *schema.Resource {
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("HPEGL_METAL_GLP_ROLE", true),
+				Description: `Field indicating the GLP role to be used, can also be set with the HPEGL_METAL_GLP_ROLE env-var`,
+			},
+			glpWorkspace: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("HPEGL_METAL_GLP_WORKSPACE", true),
+				Description: `Field indicating the GLP workspace be used, can also be set with the HPEGL_METAL_GLP_WORKSPACE env-var`,
```
Comment (mchuang3) [inline pkg/registration/registration.go:99]:
```suggestion
				Description: `Field indicating the GLP workspace to be used, can also be set with the HPEGL_METAL_GLP_WORKSPACE env-var`,
```

File: pkg/configuration/config.go (hunk 2 above)
Comment (dbozzato81) [reply pkg/configuration/config.go:222]: Nah, changed for debugging. Fixed it

**Review by mchuang3 — APPROVED** [review]

**Pull request merged**